sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy

```diff
@@ -188,4 +202,29 @@ class ProjectOverviewView extends VerticalLayout{
         return isAvailable
     }
 
+    private void enableWhenProjectIsSelected(Component component) {
+        viewModel.addPropertyChangeListener("selectedProject") {
+            Optional<ProjectSummary> projectSelection = Optional.ofNullable(viewModel.selectedProject)
+            component.enabled = projectSelection.present
+            component.visible = projectSelection.present
+        }
+    }
```

Comment:
```suggestion
    private void enableWhenProjectIsSelected(Component component) {
        viewModel.addPropertyChangeListener("selectedProject") {
            if(viewModel.selectedProject){
             component.setVisible(true)
            }else{
              component.setVisible(false)
            }
        }
    }
```

Why did you convert it to an optional? Is there an advantage that I miss? I think you can use the build in groovy check. If you use `if(viewModel.selectedProject)` it will return true if the value is not null and not empty.